### PR TITLE
Doc update core:DateTime

### PR DESCRIPTION
**Pull request by jpsiitonen**

### CODE REVIEW

File: sdk/lib/core/date_time.dart [hunk 1]
```diff
@@ -9,93 +9,110 @@ part of dart.core;
...
+/// print(dDay.isUtc); // true
 ///
-/// Use [isUtc] to determine whether a DateTime object is based in UTC.
+/// final dDayLocal = DateTime(1944, 6, 6);
+/// print(dDayLocal.isUtc); // false
+/// ```
 /// Use the methods [toLocal] and [toUtc]
 /// to get the equivalent date/time value specified in the other time zone.
+/// ```
+/// final localDay = dDay.toLocal(); // fx 1944-06-06 02:00:00.000
```
Comment (alexlindroos) [inline sdk/lib/core/date_time.dart]: fx?

File: sdk/lib/core/date_time.dart [hunk 2]
```diff
@@ -9,93 +9,110 @@ part of dart.core;
...
 /// ```
 ///
-/// To find out how much time is between two DateTime objects use
+/// To find out how much time is between two `DateTime` objects use
 /// [difference], which returns a [Duration] object:
-///
-/// ```dart
-/// var difference = berlinWallFell.difference(moonLanding);
+/// ```
+/// final difference = berlinWallFell.difference(moonLanding);
```
Comment (alexlindroos) [inline sdk/lib/core/date_time.dart:117]: should we add comment or print here?

File: sdk/lib/core/date_time.dart [hunk 3]
```diff
@@ -204,7 +223,7 @@ class DateTime implements Comparable<DateTime> {
   /// local time zone.
   ///
   /// ```dart
-  /// var thisInstant = DateTime.now();
+  /// final thisInstant = DateTime.now();
```
Comment (alexlindroos) [inline sdk/lib/core/date_time.dart]: Why not use the same variable name than before "now" ?

File: sdk/lib/core/date_time.dart [hunk 4]
```diff
@@ -622,10 +666,10 @@ class DateTime implements Comparable<DateTime> {
   /// The returned [Duration] will be negative if [other] occurs after [this].
   ///
   /// ```dart
-  /// var berlinWallFell = DateTime.utc(1989, DateTime.november, 9);
-  /// var dDay = DateTime.utc(1944, DateTime.june, 6);
+  /// final berlinWallFell = DateTime.utc(1989, DateTime.november, 9);
+  /// final dDay = DateTime.utc(1944, DateTime.june, 6);
   ///
-  /// Duration difference = berlinWallFell.difference(dDay);
+  /// final difference = berlinWallFell.difference(dDay);
```
Comment (alexlindroos) [inline sdk/lib/core/date_time.dart:675]: should we add comment or print of `difference`?

**Review by alexlindroos — CHANGES_REQUESTED** [review]

File: sdk/lib/core/date_time.dart (hunk 2 above)
Comment (jpsiitonen) [reply sdk/lib/core/date_time.dart:117]: sure.

File: sdk/lib/core/date_time.dart (hunk 1 above)
Comment (jpsiitonen) [reply sdk/lib/core/date_time.dart]: may be e.g is better..

**Review by alexlindroos — APPROVED** [review]

**Review by stepthomp — APPROVED** [review]

**Pull request closed without merging**